src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -268,13 +268,35 @@ private void readRcaEnabledFromConf() {
             String nextLine = sc.nextLine();
             rcaEnabled = Boolean.parseBoolean(nextLine);
           } catch (IOException e) {
-            LOG.error("Error reading file '{}': {}", filePath.toString(), e.getMessage());
+            LOG.error("Error reading file '{}': {}", filePath.toString(), e);
             e.printStackTrace();
             rcaEnabled = rcaEnabledDefaultValue;
           }
         });
   }
 
+  private void readAndUpdateMutesRcasDuringStart() {
+    try {
+      /* We have an edge case where both `readAndUpdateMutesRcas()` and `readAndUpdateMutesRcasDuringStart()`
+       * can try to update the muted Rca list back to back, reading rca.conf twice. This will happen when rca
+       * was turned off and then on.
+       *
+       * <p> `readAndUpdateMutesRcasDuringStart()` should only be read at the start of the process, when
+       * RCA graph is not constructed and we cannot validate the new new muted RCAs. For any other update
+       * to the muted list, the periodic rca.conf update checker will take care of it.
+       *
+       */
+      if (lastModifiedTimeInMillisInMemory == 0) {
+        Set<String> rcasForMute = new HashSet<>(rcaConf.getMutedRcaList());
+        Stats.getInstance().updateMutedGraphNodes(rcasForMute);
+        LOG.info("Updated the muted RCA Graph to : {}", rcaConf.getMutedRcaList());
+      }
+    } catch (Exception e) {
```

Comment:
Why do we need to handle exception here ? This make it quite confusing as we are not throwing any exceptions in between I assume ?

Reply:
We are only logging the exception with metric here and not throwing.